app/routes.py

```diff
@@ -1,2 +1,286 @@
-from flask import Blueprint
+import re
+from flask import Blueprint, request, make_response, jsonify
+from flask_sqlalchemy import model
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime
+import requests
+import os
+from tests.conftest import completed_task
 
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+
+@goals_bp.route("", methods=["GET", "POST"])
+def handle_goals():
+    if request.method == "GET":
+        goals = Goal.query.all()
+        goals_response = []
+        for goal in goals:
+            goals_response.append({
+                "id": goal.goal_id,
+                "title": goal.title
+            })
+
+        return jsonify(goals_response)
+
+    
+    elif request.method == "POST":
+        request_body = request.get_json()
+        if request_body.get("title"):
+            new_goal = Goal(title=request_body["title"])
+            db.session.add(new_goal)
+            db.session.commit()
+            response_value = {"goal":{
+                    "id": new_goal.goal_id,
+                    "title": new_goal.title,
+            }}
+            return make_response(response_value, 201)
+        else:
+            return make_response({"details":"Invalid data"}, 400)
+
+
+@goals_bp.route("/<goal_id>/tasks", methods=["GET", "POST"])
+def task_in_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal is None:
+        return make_response("Goal not found", 404)
+
+    if request.method == "POST":
+        request_body = request.get_json()
+        for task_id in request_body["task_ids"]:
+            task = Task.query.get(task_id)
+            task.goal_id = goal_id
+            
+
+        db.session.commit()
+        return make_response({"id":int(goal_id), "task_ids": request_body["task_ids"]}, 200)
```

Comment:
Our curriculum tends to focus on using `make_response()` for generating response data. `make_response` is a Flask function and returns `HTML`. Using this function will pass the tests for this project, however, if a client (web app) expects to receive JSON, the client would need to do extra work to access the dictionary. Or in this API, we can provide additional logic to change the headers for `make_response` to return JSON instead.

We can avoid all this extra work by either returning a dictionary (Flask turns dictionaries into JSON automatically) or by using `jsonify()`. 

So in this case you can omit `make_response` when returning a dictionary.

```suggestion
        return {"id":int(goal_id), "task_ids": request_body["task_ids"]}, 200
```